x-pack/plugins/lens/public/datasources/text_based/layerpanel.tsx

```diff
@@ -18,7 +18,10 @@ export interface TextBasedLayerPanelProps extends DatasourceLayerPanelProps<Text
 
 export function LayerPanel({ state, layerId, dataViews }: TextBasedLayerPanelProps) {
   const layer = state.layers[layerId];
-  const dataView = dataViews.indexPatternRefs.find((ref) => ref.id === layer.index);
+  let dataView = dataViews.indexPatternRefs.find((ref) => ref.id === layer.index);
+  if (!dataView) {
+    dataView = state.indexPatternRefs.find((ref) => ref.id === layer.index);
```

Comment:
Am I correct in understanding that `state.indexPatternRefs` contains only ad-hoc data views?

Reply:
No it contains everything, permanent and adhoc. I changed it to only search on this array.